src/app/reporting/ReportSchedulerImpl.cpp

```diff
@@ -59,23 +58,21 @@ void ReportSchedulerImpl::OnReadHandlerCreated(ReadHandler * aReadHandler)
     ScheduleReport(newTimeout, newNode);
 }
 
-/// @brief When a ReadHandler becomes reportable, schedule, verifies if the min interval of a handleris elapsed. If not,
-/// reschedule the report to happen when the min interval is elapsed. If it is, schedule an engine run.
+/// @brief When a ReadHandler becomes reportable, schedule, recalculate and reschedule the report.
 void ReportSchedulerImpl::OnBecameReportable(ReadHandler * aReadHandler)
 {
     ReadHandlerNode * node = FindReadHandlerNode(aReadHandler);
     VerifyOrReturn(nullptr != node);
-
     Milliseconds32 newTimeout;
     CalculateNextReportTimeout(newTimeout, node);
```

Comment:
Was already mentioned in the past, it can only return an error on the list being empty and the node not being in the list, which is impossible here as we would have returned on the VerifyOrReturn.